### PR TITLE
MINOR: Various cleanups in metadata

### PR DESCRIPTION
**Pull request by mimaison**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: metadata/src/main/java/org/apache/kafka/metadata/properties/MetaPropertiesEnsemble.java [hunk 1]
```diff
@@ -101,9 +101,7 @@ public static class Loader {
         private Optional<String> metadataLogDir = Optional.empty();
 
         public Loader addLogDirs(Collection<String> logDirs) {
```
Comment (chia7712) [inline metadata/src/main/java/org/apache/kafka/metadata/properties/MetaPropertiesEnsemble.java:103]: This method is useless, but it seems `addLogDir`  is used in for-each always. Hence, we can replace that by `addLogDirs` and then remove `addLogDir`

**Review by chia7712 — COMMENTED** [review]
@mimaison thanks for nice cleanup. PTAL one small comment

File: metadata/src/main/java/org/apache/kafka/metadata/properties/MetaPropertiesEnsemble.java (hunk 1 above)
Comment (mimaison) [reply metadata/src/main/java/org/apache/kafka/metadata/properties/MetaPropertiesEnsemble.java:103]: That's a good point! I've removed `addLogDir()`.

Overall there are a bunch of unused methods in test utilities in this module. I decided to not touch them as I know there's some work in progress in this area.

**Comment by chia7712** [comment]
@mimaison please fix the build error on scala 2.12

**Comment by mimaison** [comment]
My bad. Fixed

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**